plugins/embeddedjsp/src/main/java/org/apache/struts2/jasper/compiler/SmapUtil.java

```diff
@@ -179,14 +179,6 @@ private static String unqualify(String path) {
         return path.substring(path.lastIndexOf('/') + 1);
     }
 
-    /**
-     * Returns a file path corresponding to a potential SMAP input
-     * for the given compilation input (JSP file).
-     */
-    private static String inputSmapPath(String path) {
-        return path.substring(0, path.lastIndexOf('.') + 1) + "smap";
-    }
-
```

Comment:
Yes it also has no test coverage. I think we should either deprecate or replace those hardcopied codes with a jar dependency.

Reply:
Do we have any statistics or user feedback about the usage of `plugins/embeddedjsp`?
IMHO at least deprecation would be a good idea.

Reply:
Not the best source of truth but it shows some factor, comparing this to struts2-core shows the difference

## Embedded JSP
![Embedded JSP](https://user-images.githubusercontent.com/170103/48887273-00420380-ee2f-11e8-803a-7b255b1e18f6.png)

## Core
![Core](https://pbs.twimg.com/media/DsDo9RDWsAAcGzq.jpg)

Reply:
I think it depends. Firstly we should answer why those at all are hardcopied codes from jasper instead of a dependency. Then we should answer are they could be replaced with just a dependency easily? If so (or if it's an important plugin) we may try to replace and keep it, elsewhere, then we can deprecate it.

Reply:
I don't know the origins, why those dependencies were hardcopied instead of just use them - maybe at that time there was no other options.

Reply:
I probed and IMHO, @apache/struts-committers , @lukaszlenart yes please deprecate embeddedjsp plugin. [Servlet 3.0 already has implemented it](https://stackoverflow.com/q/5013917/1362623) so we can delete it when we upgraded to Servlet 3.0 that I think means we shouldn't spend any effort or time on it.

Reply:
We can deprecated it in Struts 2.6 and give people some time to migrate.

Reply:
I agree :+1: